src/recommendationservice_ballerina/src/recommendationservice/RecommendationService_sample_service.bal

```diff
@@ -1,13 +1,70 @@
 import ballerina/grpc;
+import ballerina/io;
+import ballerina/kubernetes;
 
-listener grpc:Listener ep = new (9090);
+// Product catalog client.
+ProductCatalogServiceBlockingClient productCat = new ("http://productcatalogservice:3550");
 
-service RecommendationService on ep {
+@kubernetes:Service {
+    serviceType: "ClusterIP",
+    name: "recommendationservice"
+}
+service RecommendationService on new grpc:Listener(8080) {
 
     resource function ListRecommendations(grpc:Caller caller, ListRecommendationsRequest value) {
-        // Implementation goes here.
+        Empty req = {};
+        // Fetch list of products from product catalog stub
+        var products = productCat->ListProducts(req);
+        if (products is grpc:Error) {
+            io:println("Error from Connector: " + products.reason() + " - "
+            + <string>products.detail()["message"]);
+
+            // You should return a ListRecommendationsResponse
+            ListRecommendationsRequest resp = {
+                user_id: value.user_id,
+                product_ids: ["9SIQT8TOJO", "6E92ZMYYFZ", "LS4PSXUNUM"]
+            };
+            io:println(resp);
+            var e = caller->send(resp);
+            e = caller->complete();
+        } else {
+            ListProductsResponse listProductResponse;
+            grpc:Headers headers;
+            [listProductResponse, headers] = products;
+            Product[] productList = listProductResponse.products;
+            
+            // Extract product id from the product list.
+            string[] productIds = [];
+            int i = 0;
+            foreach Product v in productList {
+                productIds[i] = v.id;
+                i += 1;
```

Comment:
Can replace L40-L41 with

```
productIds.push(v.id)
```

Shall we say `product` instead of `v` maybe?

Reply:
yeah this is better